software/controller/controller.sh

```diff
@@ -15,21 +30,26 @@
 # Feel free to add more checks here, but keep in mind that:
 # - They have to pass on CircleCI
 # - They have to have a very good chance of passing for other
-#   developers if they run via ./controller.sh --test
+#   developers if they run via ./controller.sh test
 
 # Fail if any command fails
 set -e
 set -o pipefail
 
 # Print each command as it executes
-set -o xtrace
+if [ -n "$VERBOSE" ]; then
```

Comment:
You can run it like this `VERBOSE=1 ./controller.sh` without permanently modifying the variable.
I really hate to add more options as then they can be at any permutation of positions. Really, all of this should be redone with `getopts` or whatever library, but maybe that's for later?

Reply:
Definitely no rush refactoring this to deal with argument permutation (especially since it is an either/or situation for most of them). The solution you propose is fine by me as well.